src/Kunstmaan/MenuBundle/Twig/MenuTwigExtension.php

```diff
@@ -18,14 +18,18 @@ class MenuTwigExtension extends \Twig_Extension
      */
     private $router;
 
+    private $repositoryClass;
+
     /**
      * @param EntityManager $em
      * @param RouterInterface $router
+     * @param $repositoryClass
      */
-    public function __construct(EntityManager $em, RouterInterface $router)
+    public function __construct(EntityManager $em, RouterInterface $router, $repositoryClass)
     {
         $this->em = $em;
         $this->router = $router;
+        $this->repositoryClass = $repositoryClass;
```

Comment:
You might want to define an interface for the repository class, so the class can be checked and an exception can be thrown if it does not implement the interface (which should contain the definition of the getMenuItemsForLanguage method).